### ISSUE TITLE
questions for training wgangp model

### ISSUE THREAD

**Issue by congyang12345** [xiumingzhang/GenRe-ShapeHD#25]
I want to train the wgangp model using my own voxel data. There are two questions bothered me. 
1. The released data you use is normalized voxel data. The element of my own voxel data is 0 or 1.  Need I normalize my voxel data? And how to normalize it? 
2. In the paper, the wgangp model was trained for 80 epoches. But the epoch was set to 1000 in the training script(https://github.com/xiumingzhang/GenRe-ShapeHD/blob/master/scripts/train_wgangp.sh). Which one is correct?
Thank you!

**Comment by ztzhang** [comment]
Hi,
1. The normalization is defined on the CAD models; we normalize each CAD model such that they fit in the unit sphere and is zero centered. We then voxelize the CAD models to produce the data.
2. The epoch number is just for max training epochs. We use the checkpoint at epoch 80 to produce the result in the paper. 
Hope this helps.

**Comment by congyang12345** [comment]
I trained the wgangp model using my own voxel data(128*128*128). At the same time, the wgangp model was trained using your released voxel data for comparison. Then I tested the naturalness work with voxel data rendered with ShapeNet objects(size: (1, 1, 128, 128, 128)). The naturalness loss offered by both discriminators is around 1500. And I  found that the generator generated the voxel model that looks like triangle. Is this wgangp model correct?

**Comment by xiumingzhang** [comment]
Could you clarify what you mean by "looks like triangle"?

**Comment by congyang12345** [comment]
When the training of wgangp model is completed using the code and the voxel data you offered, the validation results for each epoch were saved in the output directory, including several obj files named gen_voxel.obj and txt files named disc.txt. The voxel in the obj file is a triangular patch.

**Comment by xiumingzhang** [comment]
Can you send me a download link to the said .obj file?

**Comment by congyang12345** [comment]
This is part of the results saved in the output directory. https://drive.google.com/open?id=1XpTPVSzbvOt4V2_Fz2S5Q2hXsSOSxnm0

**Comment by ztzhang** [comment]
This means that your generated voxels are below the iso-surfaces threshold. 
You probably want to check the range of your generated voxels.

**Comment by congyang12345** [comment]
Thank you for your attention. I was recurring the results of your paper. The generated voxels above were based on your code and released data. The iso-surfaces threshold you set is 0.25. Do you mean the parameter is too small? What is the value you set in your paper? Hope your answer as I want to check the performance of the generator.

**Comment by ztzhang** [comment]
would you mind sending us a generated voxel file? Or can you print out the max/min value of the generated voxels?
As WGANGP training is quite standard, I feel it would be quite weird if this is actually a training problem. I've trained several WGANGP baselines using this data and it usually gives reasonable results.
@xiumingzhang any chance we can release the generator as well?

**Comment by congyang12345** [comment]
The part of the results are below. The were reproduced using your code and data. https://drive.google.com/open?id=1XpTPVSzbvOt4V2_Fz2S5Q2hXsSOSxnm0. The 0000_13_gen_vox.obj is one of the generated voxel files.

**Comment by congyang12345** [comment]
The min value of the generated voxel is infinitely close to 0, and the max value is 1.0.
Besides, I have another question. Why do you set the batch size to 4? Because the dimension of the generated voxels in the 3D shape completion module is (4,1,128,128,128)? If I want to use the trained discriminator as the naturalness loss function, and the voxel dimension I  generated is (1,1,128,128,128), should I set the batch size to 1 when I training the wgangp model?

**Comment by xiumingzhang** [comment]
The .obj you attached looks like the isosurface generated with a threshold lower than even the minimum of your voxel. To prevent the .obj file from being empty, we inserted this tiny triangle, which should be too small to see when the generated voxels are present. In your case, the isosurface is not there, so you see that triangle.
